integration/query_comparison_compat_test.go

```diff
@@ -597,8 +651,26 @@ func TestQueryComparisonCompatLte(t *testing.T) {
 
 	t.Parallel()
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/1521
 	testCases := map[string]queryCompatTestCase{
+		"Document": {
+			filter: bson.D{{"v", bson.D{{"$eq", bson.D{{"foo", int32(42)}, {"42", "foo"}, {"array", bson.A{int32(42), "foo", nil}}}}}}},
```

Comment:
Should we use "$lte" here?

Reply:
No wonder tests didn't fail :see_no_evil: thanks for catching this